modules/local/locidex_select.nf

```diff
@@ -0,0 +1,276 @@
+/*
+Locidex provides the option to select a database from a group of databases, by
+using a manifest file. To prevent copying all databases each time an allele database
+needs to be selected this file will read only the "manifest" file of the databases and
+to pick the correct allele scheme.
+
+
+Locidex Manifest is setup as:
+    {
+        db_name: [
+            {
+                "path": path/to/db/relative/to/manifest
+                # DB Config data, the newest db data will be selected as versions are not standardized
+                "config": {
+                    "db_name": "Locidex Database 1",
+                    "db_version": "1.0.0",
+                    "db_date": "yyyy-MM-dd",
+                    "db_author": "test1",
+                    "db_desc": "test1",
+                    "db_num_seqs": 53,
+                    "is_nucl": true,
+                    "is_prot": true,
+                    "nucleotide_db_name": "nucleotide",
+                    "protein_db_name": "protein"
+                }
+            }
+        ]
+    }
+*/
+
+import groovy.json.JsonSlurper
+import groovy.json.JsonBuilder
+import java.text.SimpleDateFormat
+
+
+process LOCIDEX_SELECT {
+    tag "$meta.id"
+    label "process_single"
+
+    input:
+    tuple val(meta), val(top_hit), val(contigs)
+    val manifest // This is a json file to be parsed
+    val config // optional path to a database file
+
+    output:
+    tuple val(meta), val(contigs), val(scheme), val(paired_p), emit: db_data
+    tuple val(meta), path(output_config), emit: config_data
+
+    exec:
+
+    report_name = "${meta.id}_${params.locidex.db_config_output_name}"
+    output_config = task.workDir.resolve(report_name)
+
+    if(params.allele_scheme == null && params.locidex.allele_database == null){
+        error("Allele calling is enabled but there is no allele scheme or locidex allele database location present.")
+    }
+
+    if(config){ // non null channel should evaluate to true
+        if(manifest){
+            log.warn "A database manifest file was passed along with a config file. Using the allele database specified by '--allele_scheme' "
+        }
+        // Create output summary data for a locidex allele database if an allele scheme is passed in directly
+        def jsonSlurper = new JsonSlurper()
+        String json_data = config.text
+        def locidex_config_data = jsonSlurper.parseText(json_data)
+        validate_locidex_db(locidex_config_data, params.allele_scheme)
+        write_config_data(locidex_config_data, output_config)
+        scheme = params.allele_scheme // reset the schem path to the passed allele scheme
+        paired_p = true
+        output_config
+        paired_p
+        scheme
+
+    }else{
+
+        // Tokenize the "top_hit" or species value to identify all relevant match parts of the string
+        def species_data = top_hit.split('_|\s').collect{ it.toLowerCase() }
```

Comment:
Excelelnt question! I was pondering the same thing my self. My reasoning for leaving the tag un-parsed on outputs from the mash and kraken2 processes (currently) 

1) Is to prevent having to alter the values passed to other processes in the pipeline (e.g. StarAMR) as when I make pipeline changes going forward I would like to add tests along side them. Although I think this would be a good enhancement to make going forward.

2) Previously, my thinking leaned toward the philosophy that modules should be responsible for modifying their inputs to fit the needed task. But in the context of mikrokondo, I do not think this is a necessary per-caution.

3) This would required a modification to how "species specific" QC and subtyping modules are selected for organisms. Although this may be a desired change.


I think this would be a good enhancement, so I have created an issue here: https://github.com/phac-nml/mikrokondo/issues/100